docfx_project/articles/bootstrapping-trust/part-01.encryption-everywhere.md

```diff
@@ -0,0 +1,207 @@
+piece of software. If a machine is compromised, the analogous public
+certificate will need to be untrusted on each node. To combat this
+problem, Certificate Authorities (CA) can help!
+
+#### CAs & Adding Complexity
+
+A CA enables trust deferral from multiple individual certificates to a
+single certificate. Meaning that instead of trusting each certificate,
+each piece of software will trust the CA. The CA will be used to sign
+every public certificate our software pieces need to use. How does
+"signing" work? We will cover that in parts three and four, read on.
+
+Here are the high-level steps:
+
+1. create a CA configuration via OpenSSL CNF files
+2. create the CA
+3. using the CA's public key to sign all of the public certificates
+4. distribute the CA's certificate to every machine
+5. configure the machines certificate store or configure the software
+
+For items one and two, the process can be a bit mystical. There is a
+slew of options involved in managing a CA. To perform number three, you
+will need to go through the processing of creating CSRs on behalf of the
+piece of software, and someone will have to play the role of the CA and
+resolve those CSRs. The last two steps will depend on the OS and
+software.
+
+All of these actions can all be done via a CLI or programmatically. You
+will have to spend time and energy, making sure the options are
+correctly set and learning about all the different capabilities and
+extensions. Mistakes will inevitably occur. It is time-consuming to
+debug why a specific public certificate is not working as intended. The
+tools and systems that use the certificates are purposely vague in error
+messages as not to reveal too much information to attackers.
+
+#### Further Concerns
+
+Once configured, there are still other concerns need to be taken into
+account:
+
+- What happens when systems are removed/added?
+- What happens when a certificate expires?
+- How does a system know not to trust a certificate anymore?
+- What happens when private keys need to regenerate?
+
+Remember, the systems deferred trust to the CA, which means it is
+independent of the software. CAs do not automatically handle the
+propagation of these types of events. CAs are data files on a storage
+device. Issuing or revoking certificates does not generate any kind of
+event without additional software. There is also the issue of
+certificates expiring. That "-dash 360" puts a lifetime on each
+certificate. You could make that incredibly long but exposes your system
+to having incredibly old certificates around when encryption and the
+methods to defeat it have had ample time to improve. Also, a private
+key's strength might need enhancement by generating new keys. Replacing
+private keys is necessary when with the advent of security
+vulnerabilities and improved cryptography. New private keys
+automatically require new certificates.
+
+Even if we ignore all of those concerns: who did we trust to get this
```

Comment:
the three colons in this paragraph should probably be commas.

Reply:
actually i take it back. just this one seems like it needs to be a colon - the other two seem ok. my bad